src/engraving/layout/layoutchords.cpp

```diff
@@ -113,7 +115,7 @@ void LayoutChords::layoutChords1(Score* score, Segment* segment, staff_idx_t sta
                     hasGraceBefore = true;
                 }
                 layoutChords2(c->notes(), c->up());               // layout grace note noteheads
-                layoutChords3(score->style(), c->notes(), staff, 0);              // layout grace note chords
+                layoutChords3(score->style(), chords, c->notes(), staff);              // layout grace note chords
```

Comment:
Shouldn't this particular one be `{ c }` instead of `chords`? At least that would be necessary if you want to match the behaviour that you get when collecting chords inside `layoutChords3`.

Reply:
Right, though I think it can't be list-initialized like that when passing by reference. The even better thing in that case would be to pass nothing. Or an empty vector

Reply:
No you're right, cause grace notes may be dottes as well (though I've never seen one). I'll push the fix

Reply:
For const references, you _can_ pass a "temporary value", like a list initializer, only for non-const references or pointers you'll need a variable.

Since you don't need to modify the vector inside `layoutChords3` I recommend changing the signature of `layoutChords3` to take a const reference. Only use non-const references when you _really_ need to.